src/main/java/io/vertx/redis/impl/RedisConnection.java

```diff
@@ -248,10 +248,14 @@ void send(final Command<?> command) {
         break;
       case CONNECTING:
       case ERROR:
-        pending.add(command);
+        runOnContext(v -> {
+          pending.add(command);
```

Comment:
I guess that we could avoid the `synchronized` at all if we would double check the state.

All reads/writes to the queue happen inside a `runOnContext` block, which means we're always in the same thread, it could have happen that the state changed in between the start of the `send` method and the `runOnContext`, so if on that case I'd do a double check such as:

```java
runOnContext(v -> {
  if (state.get() != CONNECTED) {
    pending.add(command);
  } else {
    // state changed so start over...
    send(command);
  }
});
```